libs/components/layout/src/lib/modules/action-button/action-button.component.html

```diff
@@ -1,89 +1,86 @@
-    "
-    class="sky-action-button sky-btn-default sky-rounded-corners"
-    tabindex="0"
-    [href]="permalink?.url"
-    [skyThemeClass]="{
-      'sky-btn sky-box': 'modern'
-    }"
-  >
-    <ng-container *ngTemplateOutlet="actionButtonContent"></ng-container>
-  </a>
-
-  <a
-    *ngSwitchCase="
-      !permalink?.route &&
-      permalink?.url &&
-      permalink?.url?.includes('://') &&
-      !permalink?.url?.startsWith('https://')
-    "
-    class="sky-action-button sky-btn-default sky-rounded-corners"
-    tabindex="0"
-    [skyHref]="permalink?.url"
-    (skyHrefChange)="onSkyHrefDisplayChange($event)"
-    [skyThemeClass]="{
-      'sky-btn sky-box': 'modern'
-    }"
-  >
-    <ng-container *ngTemplateOutlet="actionButtonContent"></ng-container>
-  </a>
-
-  <div
-    *ngSwitchDefault
-    class="sky-action-button sky-btn-default sky-rounded-corners"
-    tabindex="0"
-    role="button"
-    [skyThemeClass]="{
-      'sky-btn sky-box': 'modern'
-    }"
-    (click)="buttonClicked()"
-    (keydown.enter)="enterPress()"
-  >
-    <ng-container *ngTemplateOutlet="actionButtonContent"></ng-container>
-  </div>
+<ng-container>
+  @if (!!permalink?.route?.commands) {
+    <a
+      class="sky-action-button sky-btn-default sky-rounded-corners"
+      tabindex="0"
+      [fragment]="permalink?.route?.extras?.fragment"
+      [skyThemeClass]="{
+        'sky-btn sky-box': 'modern'
+      }"
+      [queryParams]="permalink?.route?.extras?.queryParams"
+      [queryParamsHandling]="permalink?.route?.extras?.queryParamsHandling"
+      [routerLink]="permalink?.route?.commands"
+    >
+      <ng-container *ngTemplateOutlet="actionButtonContent" />
+    </a>
+  } @else if (
+    !permalink?.route &&
+    permalink?.url &&
+    (!permalink?.url?.includes('://') || permalink?.url?.startsWith('https://'))
+  ) {
+    <a
+      class="sky-action-button sky-btn-default sky-rounded-corners"
+      tabindex="0"
+      [href]="permalink?.url"
+      [skyThemeClass]="{
+        'sky-btn sky-box': 'modern'
+      }"
+    >
+      <ng-container *ngTemplateOutlet="actionButtonContent" />
+    </a>
+  } @else if (
+    !permalink?.route &&
+    permalink?.url &&
+    permalink?.url?.includes('://') &&
+    !permalink?.url?.startsWith('https://')
+  ) {
+    <a
+      class="sky-action-button sky-btn-default sky-rounded-corners"
+      tabindex="0"
+      [skyHref]="permalink?.url"
+      [skyThemeClass]="{
+        'sky-btn sky-box': 'modern'
+      }"
+      (skyHrefChange)="onSkyHrefDisplayChange($event)"
+    >
+      <ng-container *ngTemplateOutlet="actionButtonContent" />
+    </a>
+  } @else {
+    <div
+      class="sky-action-button sky-btn-default sky-rounded-corners"
+      tabindex="0"
+      role="button"
+      [skyThemeClass]="{
+        'sky-btn sky-box': 'modern'
+      }"
+      (click)="buttonClicked()"
+      (keydown.enter)="enterPress()"
+    >
+      <ng-container *ngTemplateOutlet="actionButtonContent" />
+    </div>
+  }
```

Comment:
I feel like this would be cleaner
```
if(!!A) {}
else if (!A && !B) {
  if(!C || D) {}
  else if(C && !D) {}
}
else {}
```

Reply:
Updated. The only difference is that I did a straight `@else` instead of the `@else if` inside the second case.

```
if(!C || D) {}
else {}
```

instead of 

```
if(!C || D) {}
else if(C && !D) {}
```